File: AvaliaProf-app/src/app/shared/models/aluno.ts
import { Avaliacao } from "./avaliacao";
import { Usuario } from "./usuario";
import { Professor } from "./professor";

export class Aluno extends Usuario{
    private _curso:string;
    private _avaliacoes: Avaliacao[] = [];

    constructor(id:number,
        nome:string,
        matricula:string,
        email:string,
        senha:string,
        curso:string){
    super(id,nome,matricula,email,senha)
    this._curso = curso;
    }

    public getCurso():string{
        return this._curso;
    }

    public setCurso(curso:string):void{
        this._curso = curso;
    }

    public avaliarProfessor(professor:Professor, nota:number, comentario:string):void{
        let avaliacao:Avaliacao = new Avaliacao(nota, comentario, professor.getId(), this.getId());
        this._avaliacoes.push(avaliacao);
    }

    public getAvaliacoes():Avaliacao[]{
        return this._avaliacoes;
    }
}

File: AvaliaProf-back/src/main/java/pweb/avaliaprof/modelo/Aluno.java
package pweb.avaliaprof.modelo;
import jakarta.persistence.CascadeType;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.OneToMany;

import java.util.ArrayList;
import java.util.List;

@Entity
public class Aluno extends Usuario{
    
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    private String curso;
    @OneToMany(mappedBy = "aluno", cascade = {CascadeType.PERSIST, CascadeType.MERGE})
    private List<Avaliacao> avaliacoes = new ArrayList<Avaliacao>();

    public Aluno(){
        super();
    }
    public Aluno(String nome, String email, String senha, String matricula, String curso){
        super(nome, email, senha, matricula);
        this.curso = curso;
    }
    public String getCurso(){
        return this.curso;
    }

    public void setCurso(String curso){
        this.curso = curso;
    }

    public List<Avaliacao> listarAvaliacoes(){
        return this.avaliacoes;
    }

    public void avaliarProfessor (Avaliacao avaliacao){
        this.avaliacoes.add(avaliacao);
    }

    public void removerAvaliacao (Avaliacao avaliacao){
        this.avaliacoes.remove(avaliacao);
    }

    public void alterarAvaliacao (Avaliacao avaliacao){
        int index = this.avaliacoes.indexOf(avaliacao);
        this.avaliacoes.set(index, avaliacao);
    }

    @Override
    public String toString() {
        return "Aluno{" +
                "id=" + id +
                ", curso='" + curso + '\'' +
                ", avaliacoes=" + avaliacoes +
                '}';
    }
}
